### PR TITLE
auto save on dispose

### PR DESCRIPTION
**Pull request by sulai**
Adds some comfort to phone users: turn your phone off whenever you want. Turn it back on, resume your turn.
Also works when just moving the app to background, of course =)

### CODE REVIEW

**Comment by yairm210** [comment]
Great idea!
1 problem with the implementation though, is that on next turn, you end up cloning the GameInfo _inside_ the other thread, which is ripe for concurrency errors
The GameInfo has to be cloned before we start running concurrently on both the saving and the game being played

**Comment by sulai** [comment]
Good catch indeed! Thank you =)

File: core/src/com/unciv/logic/GameSaver.kt [hunk 1]
```diff
@@ -66,4 +66,20 @@ class GameSaver {
...
+        val gameInfoClone = gameInfo.clone()
+        kotlin.concurrent.thread {
+            // the save takes a long time (up to a second!) and we can do it while the player continues his game.
+            // On the other hand if we alter the game data while it's being serialized we could get a concurrent modification exception.
+            // So what we do is we clone all the game data and serialize the clone.
+            saveGame(gameInfoClone, "Autosave")
+
+            // do this on main thread
+            Gdx.app.postRunnable {
+                postRunnable()
```
Comment (yairm210) [inline core/src/com/unciv/logic/GameSaver.kt]: What is this for?

Comment (sulai) [reply core/src/com/unciv/logic/GameSaver.kt]: This is useful if you want to update the ui *after* the saving is done. For example, the *next turn* button in `WorldScreen` is updated that way:
```
GameSaver().autoSave(gameInfo) {
  nextTurnButton.enable() // only enable the user to next turn once we've saved the current one
  updateNextTurnButton()
}
```

**Pull request merged**